### PR TITLE
removed styles

### PR DESCRIPTION
**Pull request by mallockey**

### CODE REVIEW

File: src/components/PlacesAutocomplete.tsx [hunk 1]
```diff
@@ -21,24 +21,24 @@ const PlacesAutocomplete: React.FC = () => {
...
-    const listElements = document.getElementsByTagName('li')
+    const listElements = document.getElementsByTagName('button')
     if (listElements.length === 0) {
       return null
     }
     return Array.from(listElements)
   }
 
-  function clearLastSuggestion() {
+  function clearLastSuggestion(): void {
```
Comment (Penspinner) [inline src/components/PlacesAutocomplete.tsx:31]: No need to type function return values unless needed. Leverage TypeScript's type inference.

```suggestion
  function clearLastSuggestion() {
```

File: src/components/PlacesAutocomplete.tsx [hunk 2]
```diff
@@ -21,24 +21,24 @@ const PlacesAutocomplete: React.FC = () => {
...
   }
 
   function clearAllSuggestions() {
     const listElements = getSuggestionsElements()
-    listElements.forEach((item) => (item.className = ''))
+    listElements ? listElements.forEach((item) => (item.className = '')) : ''
   }
 
-  function handleHover(event) {
+  function handleHover(event: React.ChangeEvent<HTMLInputElement>): void {
```
Comment (Penspinner) [inline src/components/PlacesAutocomplete.tsx:41]:
```suggestion
  function handleHover(event: React.ChangeEvent<HTMLInputElement>) {
```

File: src/components/PlacesAutocomplete.tsx [hunk 3]
```diff
@@ -50,13 +50,9 @@ const PlacesAutocomplete: React.FC = () => {
     }
   }
 
-  function handleKeyDown(event) {
+  function handleKeyDown(event: KeyboardEvent): void {
```
Comment (Penspinner) [inline src/components/PlacesAutocomplete.tsx:53]:
```suggestion
  function handleKeyDown(event: KeyboardEvent) {
```

File: src/components/PlacesAutocomplete.tsx [hunk 4]
```diff
@@ -82,11 +78,15 @@ const PlacesAutocomplete: React.FC = () => {
     }
   }
 
-  function handleInput(e) {
-    setValue(e.target.value)
+  function handleInput(event: React.ChangeEvent<HTMLInputElement>): void {
```
Comment (Penspinner) [inline src/components/PlacesAutocomplete.tsx:81]:
```suggestion
  function handleInput(event: React.ChangeEvent<HTMLInputElement>) {
```

File: src/components/PlacesAutocomplete.tsx [hunk 5]
```diff
@@ -82,11 +78,15 @@ const PlacesAutocomplete: React.FC = () => {
...
+    setValue(event.target.value)
+  }
+
+  interface SubmitBuildingProps {
+    description: 'string'
   }
 
-  async function submitBuilding({ description }) {
+  async function submitBuilding({ description }: SubmitBuildingProps): Promise<void> {
```
Comment (Penspinner) [inline src/components/PlacesAutocomplete.tsx:89]:
```suggestion
  async function submitBuilding({ description }: SubmitBuildingProps) {
```

**Review by Penspinner — CHANGES_REQUESTED** [review]

**Comment by Penspinner** [comment]
You should not be using `document` methods to query for the buttons. A representation of the database values should be stored in state, and you should be looping through that state to render the buttons. You could use `react-query` to have that abstracted out for you. This could be done in a separate PR though.

**Comment by mallockey** [comment]
> You should not be using `document` methods to query for the buttons. A representation of the database values should be stored in state, and you should be looping through that state to render the buttons. You could use `react-query` to have that abstracted out for you. This could be done in a separate PR though.

But the data is not from the database it's from googles places autocomplete API. Does this still apply?

File: src/components/PlacesAutocomplete.tsx [hunk 6]
```diff
@@ -30,33 +30,35 @@ const PlacesAutocomplete: React.FC = () => {
 
   function clearLastSuggestion() {
     const listElements = getSuggestionsElements()
-    listElements[highLightedIndex].className = ''
+    listElements ? (listElements[highLightedIndex].className = '') : ''
```
Comment (Penspinner) [inline src/components/PlacesAutocomplete.tsx:33]:
```suggestion
    listElements?.[highLightedIndex].className = ''
```

File: src/components/PlacesAutocomplete.tsx [hunk 7]
```diff
@@ -30,33 +30,35 @@ const PlacesAutocomplete: React.FC = () => {
...
+  function clearAllSuggestions(selectedIndex: number) {
     const listElements = getSuggestionsElements()
-    listElements.forEach((item) => (item.className = ''))
+    if (listElements) {
+      listElements.forEach((item, currentIndex) => {
+        if (selectedIndex !== currentIndex) {
+          item.className = ''
+        }
+      })
+    }
```
Comment (Penspinner) [inline src/components/PlacesAutocomplete.tsx:44]:
```suggestion
    listElements?.forEach((item, currentIndex) => {
      if (selectedIndex !== currentIndex) {
        item.className = ''
      }
    })
```

**Review by Penspinner — CHANGES_REQUESTED** [review]
Don't remove `package-lock.json`. We need that for npm.

**Pull request merged**